system-int-test/test/cmr/system_int_test/bootstrap/bulk_index/generics_test.clj

```diff
@@ -192,3 +196,353 @@
        (core/reenable-automatic-indexing)))))
+
+(deftest ^:oracle bulk-index-generics-association
+  (testing "Bulk index two grids which have been associated together and contain association details"
+    (system/only-with-real-database
+  ;; Disable message publishing so items are not indexed but, they are ingested into the system.
+  (core/disable-automatic-indexing)
+  ;; Create two grids
+     (let [token (echo-util/login (system/context) "user1")
+           grid1 (generic/ingest-generic-document nil "PROV1" "tg1" :grid generic/grid-good :post)
+           grid2 (generic/ingest-generic-document nil "PROV1" "tg2" :grid generic/grid-good :post)
+           grid1-concept-id (:concept-id grid1)
+           grid2-concept-id (:concept-id grid2)
+           grid1-revision-id (:revision-id grid1)
+           grid2-revision-id (:revision-id grid2)]
+       ;; No index since they are disabled
+       (is (= 0 (:hits (search/find-refs :grid {}))))
+       ;; bulk index to create the indexes for the ingested grids
+       (bootstrap/bulk-index-grids "PROV1")
+       (index/wait-until-indexed)
+       ;; Grids should now be indexed
+       (is (= 2 (:hits (search/find-refs :grid {}))))
+       ;; Associate the two grids
+       (let [response1 (association-util/generic-associate-by-concept-ids-revision-ids
+                        token grid1-concept-id grid1-revision-id [{:concept-id grid2-concept-id :data "some data" :revision-id grid2-revision-id}])
+             _ (index/wait-until-indexed)
+             grid-search-result (association-test/get-associations-and-details "grids.umm_json" "native_id=tg1" :grids true)])
+       ;; Associations in the index should be empty since they were not there when the record was re-indexed
+       (let [grid-search-pre-bulk (association-test/get-associations-and-details "grids.umm_json" "native_id=tg1" :grids true)]
+         (is empty? (set (:association-details grid-search-pre-bulk))))
+       ;; re-index the association since it was not there at the start of the test
+       (bootstrap/bulk-index-grids "PROV1")
+       (index/wait-until-indexed)
+       ;; Should still have two indexes for the grids
+       (is (= 2 (:hits (search/find-refs :grid {}))))
+       ;; The association should still be indexed
+       (let [grid-search-result2 (association-test/get-associations-and-details "grids.umm_json" "native_id=tg1" :grids true)]
+         (is (= (set [{:data "some data" :concept-id grid2-concept-id :revision-id grid2-revision-id}])
+                (set (:association-details grid-search-result2))))))
+    ;; Reenable message publishing
+     (core/reenable-automatic-indexing))))
+
+(deftest ^:oracle bulk-index-generics-association-json-endpoint
+  (testing "Bulk index two grids which have been associated together and contain association details search on json endpoint"
+    (system/only-with-real-database
+  ;; Disable message publishing so items are not indexed but, they are ingested into the system.
```

Comment:
spacing?

Reply:
Fixed comment spacing to put on same line